service/src/main/java/gov/nasa/pds/api/registry/controller/SwaggerJavaTransmuter.java

```diff
@@ -103,6 +105,29 @@ protected ResponseEntity<Object> processs(EndpointHandler handler, URIParameters
     }
   }
 
+  protected ResponseEntity<Map<String,Object>> processHealthcheck() {
+    long begin = System.currentTimeMillis();
+    try {
+      HttpStatus responseStatus = HttpStatus.OK;
+      HealthcheckLogic hcLogic = new HealthcheckLogic(this);
+      Map<String, Object> response = hcLogic.healthcheck();
+
+      if ((boolean) response.get(HealthcheckLogic.FAILURES_PRESENT)) {
+        responseStatus = HttpStatus.I_AM_A_TEAPOT;
```

Comment:
Following on from previous comment, if we *can't* do it properly, a comment explaining what is being done here and why would be helpful to $futureDev

Reply:
Good point, comment added.